integrations/sonarqube/integration.py

```diff
@@ -0,0 +1,173 @@
+    cwe: list[str] | None
+    impact_severities: list[Literal["HIGH", "LOW", "MEDIUM"]] | None = Field(
+        alias="impactSeverities"
+    )
+    impact_software_qualities: list[
+        Literal["MAINTAINABILITY", "RELIABILITY", "SECURITY"]
+    ] | None = Field(alias="impactSoftwareQualities")
+    statuses: list[
+        Literal["OPEN", "CONFIRMED", "FALSE_POSITIVE", "ACCEPTED", "FIXED"]
+    ] | None
+    languages: list[str] | None
+    owasp_asvs_level: Literal["1", "2", "3"] | None = Field(alias="owaspAsvsLevel")
+    resolved: Literal["yes", "no", "true", "false"] | None
+    rules: list[str] | None
+    scopes: list[Literal["MAIN", "TESTS"]] | None
+    sonarsource_security: list[str] | None = Field(alias="sonarsourceSecurity")
+    tags: list[str] | None
+
+    def generate_request_params(self) -> dict[str, Any]:
+        value = self.dict(exclude_none=True)
+        if assigned := value.pop("assigned", None):
+            value["assigned"] = assigned
+        if assignees := value.pop("assignees", None):
+            value["assignees"] = ",".join(assignees)
+        if clean_code_attribute_categories := value.pop(
+            "clean_code_attribute_categories", None
+        ):
+            value["cleanCodeAttributeCategories"] = ",".join(
+                clean_code_attribute_categories
+            )
+        if code_variants := value.pop("code_variants", None):
+            value["codeVariants"] = ",".join(code_variants)
+        if created_before := value.pop("created_before", None):
+            value["createdBefore"] = created_before
+        if created_after := value.pop("created_after", None):
+            value["createdAfter"] = created_after
+        if cwe := value.pop("cwe", None):
+            value["cwe"] = ",".join(cwe)
+        if impact_severities := value.pop("impact_severities", None):
+            value["impactSeverities"] = ",".join(impact_severities)
+        if impact_software_qualities := value.pop("impact_software_qualities", None):
+            value["impactSoftwareQualities"] = ",".join(impact_software_qualities)
+        if statuses := value.pop("statuses", None):
+            value["statuses"] = ",".join(statuses)
+        if languages := value.pop("languages", None):
+            value["languages"] = ",".join(languages)
+        if owasp_asvs_level := value.pop("owasp_asvs_level", None):
+            value["owaspAsvsLevel"] = owasp_asvs_level
+        if resolved := value.pop("resolved", None):
+            value["resolved"] = resolved
+        if rules := value.pop("rules", None):
+            value["rules"] = ",".join(rules)
+        if scopes := value.pop("scopes", None):
+            value["scopes"] = ",".join(scopes)
+        if sonarsource_security := value.pop("sonarsource_security", None):
+            value["sonarsourceSecurity"] = ",".join(sonarsource_security)
+        if tags := value.pop("tags", None):
+            value["tags"] = ",".join(tags)
```

Comment:
This looks a bit messy, is there a way to use `switch case` or something to make the code more readable?